packages/relay/src/lib/clients/mirrorNodeClient.ts

```diff
@@ -886,6 +919,31 @@ export class MirrorNodeClient {
       1,
       MirrorNodeClient.mirrorNodeContractResultsLogsPageMax,
     );
+
+    for (const log of logResults) {
+      if (log && (log.transaction_index == null || log.block_number == null || log.index == null)) {
+        console.log(logResults);
+        if (this.logger.isLevelEnabled('debug')) {
+          this.logger.debug(
+            `${requestDetails.formattedRequestId} Contract result log contains undefined transaction_index, block_number, or index. Retrying after a delay of ${shortDelay} ms.`,
```

Comment:
From this log we won't know which field is undefined

Reply:
You're right that would give more info. Added those fields and an extra field for transaction hash for easier debug.